### ISSUE TITLE
Unable to replicate paper results on GLUE RTE task using example implementation

### ISSUE THREAD

**Issue by dwz99999** [AnswerDotAI/ModernBERT#225]
Dear maintainers,

I am currently using ModernBERT for my research project. However, while doing baseline runs, I found that the model performed poorly during fine-tuning. I was able to replicate this using the example notebook with minimal modifications and hyperparameters provided in the paper, where training on RTE produced accuracy in the range of **57-66%** as opposed to 87.4% claimed in the paper. Because the large discrepancy, I strongly suspect there may be an issue with the HuggingFace implementation or the checkpoint.

I have created a repository containing the minimal amount of code to replicate this issue: [https://github.com/dwz99999/modernbert-glue-rte](https://github.com/dwz99999/modernbert-glue-rte). I would greatly appreciate any help with this issue.

# Expected behaviour

Fine-tuning ModernBERT-base achieves an accuracy of close to 87.4% on RTE, as claimed in [the paper in Table 5](https://arxiv.org/pdf/2412.13663).

# Bug description

Fine-tuning ModernBERT on the GLUE RTE task using the example notebook produces accuracy scores of 57-66%, significantly lower **(by more than 30%)** than accuracy achieved in the paper.

# Steps to replicate

1. Install the following packages:
```
torch==2.6.0
pandas==2.2.3
scikit-learn==1.6.1
transformers==4.51.2
datasets==3.5.0
flash-attention==1.0.0
accelerate==1.6.0
ipykernel==6.29.5
```
2. In [examples/finetune_modernbert_on_glue.ipynb](https://github.com/AnswerDotAI/ModernBERT/blob/main/examples/finetune_modernbert_on_glue.ipynb), change `task` to `"rte"`
3. In the notebook, change hyperparameters to:
```
train_bsz, val_bsz = 32, 32
lr = 5e-5
betas = (0.9, 0.98)
n_epochs = 3
eps = 1e-6
wd = 1e-5
```
These are the exact hyperparameters used in [the paper](https://arxiv.org/pdf/2412.13663) for the RTE task, as shown in Table 6.
4. Remove `compile=False` from the `AutoModelForSequenceClassification.from_pretrained` call: see [Issue 190](https://github.com/AnswerDotAI/ModernBERT/issues/190)
5. Run the notebook - expected result is ~87.4% (result claimed in the paper). However in practice the accuracy score is only able to reach 57-65% after 3 epochs.

# Additional information

Code to replicate available on GitHub:
[https://github.com/dwz99999/modernbert-glue-rte](https://github.com/dwz99999/modernbert-glue-rte)

Accuracy over 5 runs:

- 0.653430
- 0.574007
- 0.588448
- 0.617329
- 0.664260

System configuration:
RTX 3060 12GB
NVIDIA driver 566.03
CUDA 12.7
WSL Ubuntu 24.04 LTS

By default, the HuggingFace implementation uses mean pooling for sequence classification. However, setting `hf_model.config.classifier_pooling="cls"` before the training run also does not appear to improve performance.

I have also tested on Colab, with a Tesla T4 instance, to the same result

![Image](https://github.com/user-attachments/assets/3f14f9af-1a71-4ed6-a6bf-c6c8229bef0b)

**Comment by NohTow** [comment]
Hey,
As stated in the paper:
> RTE MRPC and STS-B checkpoints are trained starting from the MNLI checkpoint.

I believe this is not the case of the notebook.

**Comment by dwz99999** [comment]
Missed that section of the paper, thanks so much
